File: Data Structures & Algorithms/minimum-stack/submission-8.py
class MinStack:

    def __init__(self):
        self.stack = []
        
    def push(self, val: int) -> None:
        self.stack.append(val)
        

    def pop(self) -> None:
        self.stack.pop(-1)

    def top(self) -> int:
        return self.stack[len(self.stack)-1]

    def getMin(self) -> int:
        mval = 2147483647
        for d in self.stack:
            if d < mval:
                mval = d

        return mval

File: Data Structures & Algorithms/validate-parentheses/submission-8.cpp
class Solution {
public:
    bool isValid(string s) {
        std::stack<char> stk;
        unordered_map<char, char> closeToOpen{
            {')', '('},
            {']', '['},
            {'}', '{'}
        };

        for(auto& c:s){
            if(closeToOpen.count(c) && !stk.empty() && stk.top() == closeToOpen[c]){
                stk.pop();
            } else{
                stk.push(c);
            }
        }

        return stk.empty();
    }
};
